test/controllers/contact/IsOfficialAddressControllerSpec.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.contact
+
+import base.SpecBase
+import controllers.actions.{AuthIdentifierAction, FakeAuthIdentifierAction}
+import forms.contact.IsOfficialAddressFormProvider
+import models.NormalMode
+import navigation.ContactNavigator
+import navigation.FakeNavigators.FakeContactNavigator
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, verify, _}
+import org.scalatest.BeforeAndAfterEach
+import pages.contact.IsOfficialAddressPage
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.Helpers._
+import repositories.UserAnswerRepository
+import views.html.contact.IsOfficialAddressView
+
+import scala.concurrent.Future
+
+class IsOfficialAddressControllerSpec extends SpecBase with BeforeAndAfterEach {
+
+  override lazy val userAnswers = Some(emptyUserAnswers)
+
+  override def applicationBuilder(): GuiceApplicationBuilder =
+    new GuiceApplicationBuilder()
+      .overrides(
+        bind[UserAnswerRepository].toInstance(mockUserAnswerRepository),
+        bind[ContactNavigator].toInstance(FakeContactNavigator),
+        bind[AuthIdentifierAction].to[FakeAuthIdentifierAction]
+      )
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockUserAnswerRepository)
+  }
+
+  val view: IsOfficialAddressView = injector.instanceOf[IsOfficialAddressView]
+  val formProvider: IsOfficialAddressFormProvider = injector.instanceOf[IsOfficialAddressFormProvider]
```

Comment:
Modify this to inject[IsOfficialAddressFormProvider]

Reply:
modified